.github/workflows/ci.yaml

```diff
@@ -11,8 +11,99 @@ on:
   pull_request:
 
 jobs:
-  test:
-    name: Test Rails
-    uses: alphagov/govuk-infrastructure/.github/workflows/test-rails.yaml@main
+  cache-dependencies:
+    name: Cache dependencies
+    uses: alphagov/govuk-infrastructure/.github/workflows/cache-dependencies.yaml@main
+
+  precompile-assets:
+    name: Precompile Rails assets
+    needs: cache-dependencies
+    uses: alphagov/govuk-infrastructure/.github/workflows/precompile-rails-assets.yaml@main
```

Comment:
If we're testing the ability to pre-compile assets, we'll want to run it against the production environment so that it tests compression - which is where people get caught out. 

May be good to clarify if this is an optimisation step or a test. For what it's worth I'd imagine it might not add a huge amount of value as an optimisation - maybe only needed for an app with exceptionally slow asset compilation (Whitehall was once in that way but I think it's ok now)

Reply:
Yeah, this job was meant to serve as a test for precompiling, but mainly was to cache the assets if they're re-used in multiple subsequent steps. (However, turns out that doesn't save any time.) I've removed this as an individual job and kept it as a step in the test run jobs (if the app has assets). At the moment hat will be compiling them with `RAILS_ENV=test`, but that could be changed [in the composite action.](https://github.com/alphagov/govuk-infrastructure/blob/add91b7629162fb3ff27b6aaebf558260b0d8fe0/.github/actions/precompile-rails-assets/action.yml#L18). 

We might want to move this to separate job, as it'll be easier to identify which bit of CI is failing and to not block the tests running if they take a long time to compile. Will look at that outside the initial PRs.